src/command_modules/azure-cli-interactive/azclishell/app.py

```diff
@@ -494,92 +481,53 @@ def _special_cases(self, text, cmd, outside):
-                            skip = False
-
-                            for counter in range(cmd_base.count(SELECT_SYMBOL['query'])):
-                                if counter < len(results) and res_counter < len(results[counter]):
-                                    try:
-                                        base = base.replace(
-                                            SELECT_SYMBOL['query'], results[counter][res_counter], 1)
-                                    except TypeError:  # because of bad input
-                                        pass
-                                else:  # if there aren't an even number of results, skip it
-                                    skip = True
-                            if not skip:
-                                self.cli_execute(base)
-                    else:
-                        for res_counter, _ in enumerate(results[0]):
-                            base = cmd_base
-                            skip = False
-
-                            for counter in range(cmd_base.count(SELECT_SYMBOL['query'])):
-                                if counter < len(results) and res_counter < len(results[counter]):
-                                    try:
-                                        base = base.replace(
-                                            SELECT_SYMBOL['query'], results[counter][res_counter], 1)
-                                    except TypeError:  # because of bad input
-                                        pass
-                                else:  # if there aren't an even number of results, skip it
-                                    skip = True
-                            if not skip:
-                                self.cli_execute(base)
-                    continue_flag = True
-
+            if len(args) == 1:  # if arguments start with query_symbol, just print query result
+                if args[0] == query_symbol:
+                    print(json.dumps(self.last.result, sort_keys=True, indent=2), file=self.output)
+                elif len(args) == 1 and args[0].startswith(query_symbol):
+                    query = args[0][symbol_len:]
+                    print(json.dumps(jmespath.search(query, input_dict), sort_keys=True, indent=2), file=self.output)
+            else:  # query, inject into cmd
+                def sub_result(arg):
+                    escaped_symbol = re.escape(query_symbol)
+
+                    def jmespath_query(match):
+                        query_result = jmespath.search(match.group(0)[symbol_len:], input_dict)
+                        if isinstance(query_result, list):
+                            return ' '.join(query_result)
+                        elif isinstance(query_result, (six.text_type, six.string_types)):
+                            return query_result
+                        else:  # query result is not a string or list, return invalid query message
+                            raise CLIError
```

Comment:
Should raise an exception that includes a message of what went wrong.
If you do this, you then won't need the comment above explaining what the exception is for.
Then also the user can see why it failed.

Reply:
true, this is no longer there after some recent changes I made in the code, but I am now printing error messages when I catch them